cli/src/cli.rs

```diff
@@ -54,6 +54,11 @@ pub enum Subcommand {
 	#[structopt(name = "benchmark", about = "Benchmark runtime pallets.")]
 	Benchmark(frame_benchmarking_cli::BenchmarkCmd),
 
+	/// Compiles a sample wasm code in order to measure the machine capabilities
+	/// of running PVF host.
+	#[cfg(not(debug_assertions))]
```

Comment:
Including this only conditionally makes it less usable, since there is no information of what is happening. Erroring out when invoked seems to be the better usability perspective, and letting the user know why it happened.

Reply:
We'll still have to `[cfg(not(debug_assertions))]` this check on the first validator run though to avoid guaranteed error

Reply:
I do not understand? This is at compile time.


What I mean is

```rust
#[cfg(debug_assertions)]
{
bail!()
}
#[cfg(not(debug_assertions))]
{
// run bench
}
```

rather than just dropping the subcommand.

Reply:
From #4196
> Consideration:
>Running this check by default on the first run and bail if insufficient might make sense.

If this run happens in debug mode it makes sense to me to disable this check during compile time, right?

Reply:
I'd recommend to take the perspective of the unexpecting persion compiling the binary, and afterwards trying to run the benches.
`No such subcommand` is not-helpful in this context, an error `This subcommand only works in release mode, please re-compile` gives clear instructions on how to resolved the issue.

Reply:
I agree with a subcommand usage, but if you want to debug a validator and get "please re-compile" because this check runs on the first launch, this is not convenient either

Reply:
My idea was to only fail on the explicit call of the subcommand, not on the implicit one. Hence why the comment was at the enum variant representing the subcommand, and not at the impl :)